geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -158,16 +158,25 @@ object FilterHelper {
 
   def isFilterWholeWorld(f: Filter): Boolean = f match {
       case op: BBOX       => isOperationGeomWholeWorld(op)
-      case op: Within     => isOperationGeomWholeWorld(op)
       case op: Intersects => isOperationGeomWholeWorld(op)
       case op: Overlaps   => isOperationGeomWholeWorld(op)
-      case op: Contains   => isOperationGeomWholeWorld(op)
+      case op: Within     => isOperationGeomWholeWorld(op, SpatialOpOrder.PropertyFirst)
+      case op: Contains   => isOperationGeomWholeWorld(op, SpatialOpOrder.LiteralFirst)
```

Comment:
This solution looks great.  

I think it might be worthwhile to have a unit test for this, but I wouldn't require it.